File: javascript/024.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
  let current = head;
  while(true) {
    const first = current;
    if (first === null) return head;
    const second = current.next;
    if (second === null) return head;
    const temp = first.val;
    first.val = second.val;
    second.val = temp;

    current = current.next.next;
  }
  return head;
};

File: ruby/024.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val)
#         @val = val
#         @next = nil
#     end
# end

# @param {ListNode} head
# @return {ListNode}
def swap_pairs(head)
  return head if head.nil?

  first = head.next
  return head if first.nil?

  second = swap_pairs(first.next)

  first.next = head
  head.next = second

  first
end